### DISCUSSION TITLE
Welcome to Ballistica Discussions!

### DISCUSSION THREAD

**Discussion by efroemling** [efroemling/ballistica#219] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Dliwk** [comment]
## :wave: Hello everyone!
I'm a young 15 years old (as in 2020) student from Russia who loves fruits and often does some random stuff on my laptop, from chatbots to a (very) minimalistic OS kernel. The last one by the way never finished :sob:

I think it's fun to communicate with other members of this community, because there are common interests and at least for the reason that some of them live on the other half of the planet :earth_asia:

**Comment by kalenk** [comment]
## 👋 Hello!
I'm a 15 year old guy from Russia who spend part of the free time at the PC: learning programming, playing games, developing in video editing, etc...

I would like to return to the days when my friends and I were going to the party to play BombSquad 😔. So i want to wish Developers forces to develop a pumped-up BombSquad💪

**Comment by ghost** [comment]
Hello everyone, I am 15 y.o. too. Programming is one of my hobbies.
BombSquad is my favourite game so that's why I am here.

Nice to meet you all

**Comment by YashKandalkar** [comment]
## Hello!

I am a 18 y/o programmer. I do web development mostly with React but python is the first language I've learnt. I go through the code base of ballistica when I'm free from college and would definitely like to contribute!

I have been playing BombSquad on PC for 3-4 years now and I still play it daily with my frens :)

**Comment by AllStarYT** [comment]
Well, hi! I'm 15 old brazillian and you know... I just like the game I'm not a programmer or something like that, I'm just pretty curious about the game, still playing since 2014, when the game was launched at Google Play!

**Comment by PythonCode-blip** [comment]
Hello, I'm a 16 year old student from Mexico (Veracruz) and I played
BombSquad since 2013!

**Comment by efroemling** [comment]
Welcome everybody, and good to meet you! Thanks for taking an interest in the project..
A quick bit of history about myself and BombSquad (or Ballistica I should say):

I live in San Francisco, California (though I grew up in Illinois).

I started coding and trying to write games when I was about 15. BombSquad started as a hobby project towards the end of my College years, though it reused some bits of code that I had written years before for other projects. At the time, my roommates and I would often have big groups of friends over to play LAN games of Halo or other local multiplayer stuff like Smash Bros or BomberMan, so I started writing BombSquad as something we could slot in and play between other games in these situations. This was around 2004 or so. 

After college, I started working at Pixar doing visual effects for their movies, which sucked up most of my attention, but I would also still tinker on BombSquad occasionally in my spare time. It was a good way to learn Python (which is used a lot in film and visual effects work) as well as experiment with other tech such as rigid body dynamics/etc. A lot of this knowledge wound up being useful in my work at Pixar. For example, if you've seen the movie 'Up', the balloon simulations owe a lot to things I learned while writing BombSquad.

For years BombSquad remained only a hobby, but around 2010 I was inspired to clean the game up and release it by a few friends of mine who had achieved success writing early games for the iPhone. So in 2011 I launched it on the Mac App Store. It was successful enough there that I decided to port it to more platforms and eventually decided to leave Pixar to focus on games full-time. So that's what I've been up to since 2013. I spent a few years working with friends of mine on some of their games (The Zombie Gunship series), but now I am 100% focused on BombSquad.

I feel extremely fortunate that the project has done well enough to allow me to keep working on it like this, and I have a lot of plans for its future. Working as a solo developer can be slow going, but it is also rewarding because one can try to make the exact thing they want to make. (well, as long as it continues to pay the bills). It's also awesome to see folks like you taking an interest in it, and I hope its something you can all learn from or customize or have fun with. I was very much inspired by playing Warcraft 3 mods back in the day and seeing all the creative things people did with them (which wound up spawning entire genres such as MOBAs). I've been amazed several times seeing mods people have made for BombSquad and I'm very inspired to try and make things like that easier and easier to do in the future.

Anyway, enough rambling about me; I should get back to coding. A huge amount of my time this past year or two has been spent on cloud stuff; getting the game ready to be a modern online experience with things like friend lists, matchmaking, etc. as we march towards 2.0.  Hopefully I will have some cool stuff on that front to show soon...

Cheers all,
-Eric

**Reply by AllStarYT** [reply]
What a great history XD, and I actually saw your name in the Toy Story 3 credits, and I was like "wtf eric froemling" lol

Well, thanks for share your history! Is good to talk with the developer of a game that made hours of my life much more fun!

**Reply by ghost** [reply]
Awesome, I quit playing BS a year ago and I was having fun coding some mods for the game. I guess this is the right time to go back again, thank you for this game, I enjoy playing this amazing game with my friends.

**Reply by MrStahlfelge** [reply]
Hi, 39 y/o developing since 13. Sorry, I can't support the 15 y/o or developing since 15 community. :D

I play Bomsquad for 4 years know, and I am glad the game suceeded. I first saw it on Google Play, but play it on desktop usually. A Steam release would be very fortunate.

Bombsquad is very well done (I am saying this as a dev myself), props for it. From a user's perspective, mostly a better multiplayer with matchmaking and a moderation system would be needed. I understand this was not the games focus when you developed it at first, but it is highly needed nowadays. 

Keep up the good work!

**Reply by efroemling** [reply]
Thanks, and nice to see I'm not the only older person here 😅 (though its awesome to have a variety of ages)

And yeah getting online multiplayer working well is my #1 focus. Most of the game's evolution was me playing local multiplayer with groups of 6 or 8 friends on a big TV, but now in the era of online multiplayer (not to mention Covid) my focus has shifted to trying to capture that same feeling with online play. It's nowhere near there yet but I've been doing lots of work on cloud servers, matchmaking, etc these past few years which will hopefully start paying dividends soon.

**Comment by PythonCode-blip** [comment]
Hey Eric! Just wondering, when will there be an Official BombSquad Discord
Community, it'll be a great way to connect with players all around the
world. It's fine if you disagree with me but I think it's a great idea,
thank you for reading this 😀

**Reply by AllStarYT** [reply]
That would be really cool!

**Reply by efroemling** [reply]
I agree it would be a good thing, and I've gotten asked about that a lot. I'm just often in heads-down mode so I worry it would suffer from a lack of moderation/attention on my part. Though perhaps I could wrangle some volunteer moderators/organizers to help out...

**Reply by AllStarYT** [reply]
Yeah, it has good and bad points, like the atention that you would have to not turn everything into fire XD, in contrast today discord is one of the best (or the best one) "services" to unite communities like BombSquad

Well, let's wait and see 👀

**Comment by xapyku** [comment]
Good afternoon.
Thank you for the game.
Deal with the cheaters, otherwise the game will die...
PS. Oh, and the exit vote sucks. Only you win fairly - you get thrown out.

**Comment by TheNotEasy** [comment]
Hello! I'm 12 years old from Russia.
Before I learned about modding in bombsquad, I was familiar with python and knew it quite well. Once when I was playing bombsquad with friends, I studied the settings. And I found a modding guide there. I was interested in this and wanted to know more about it, and what my surprise was that the logic of this game was written in Python. I was still more interested. And now I'm writing my first game, but I really wanted the documentation to be a little more.

Sorry for bad English. I used a translator

**Comment by DavidReadsGitHub** [comment]
**Hi!**

I’m just 13 years old and lives in the Philippines.
Well as you can see, I’m the main driving force behind the Filipino Translation of BombSquad.
I’ve been a fan of the game since 2016 and my friends’ ambition with the game was quite low by they played with me sometimes. Though I’m a Filipino, my grammar and my knowledge  of it’s language is a “meh” so I use Google Translate but edited some of the sentence/words/names to fit the bill.

The first time I played BombSquad was when I bought “Kata” (A small console-based company that was supported with Android). At first I wasn’t interested because I was mindlessly using a character with only a Right stick and the buttons in the TV to click on but when I press one of its buttons on the controller, it was a breakthrough and started playing the game more and more frequently. And that’s where I am today!

I have a lot of childhood games over the years and BombSquad was one of them. For now, my interest in the game has returned once again. 

Thank you!

**Comment by IvanPragma** [comment]
## 👋Hello
My name is Ivan, I am 16 years old, I'm from Russia. I've been playing BombSquad for probably 5 years now. I have my own YouTube channel and servers (FmS).

**Reply by RyanAuxil24** [reply]
@IvanPragma , can you tell your youtube channel name?

**Reply by Dliwk** [reply]
> @IvanPragma , can you tell your youtube channel name?
> 

Ms Company or something like that... I'm not sure, you better wait for him to answer :)

**Reply by RyanAuxil24** [reply]
@IvanPragma 

![Screenshot (22)](https://user-images.githubusercontent.com/99529437/170975805-7ebde4b5-78b2-43da-89c1-1cad63268224.png)


is this your youtube channel name, " Ms Company - BombSquad" ?

**Reply by RyanAuxil24** [reply]
@Dliwk , is it this " Ms Company - BombSquad" ?

**Reply by Dliwk** [reply]
> @Dliwk , is it this " Ms Company - BombSquad" ?

Looks like that

**Reply by RyanAuxil24** [reply]
ok bro

**Reply by IvanPragma** [reply]
> @IvanPragma , can you tell your youtube channel name?

Yes, "MS company - BombSquad" https://www.youtube.com/channel/UCnukeLrmoOqRfD4KWokdyvA

**Comment by RyanAuxil24** [comment]
Hello there.. , @efroemling, when will you release the update BombSquad 1.7 in Google Play Store?

I am waiting for it....

**Reply by efroemling** [reply]
> Hello there.. , @efroemling, when will you release the update BombSquad 1.7 in Google Play Store?
> 
> I am waiting for it....

As soon as I can. I need to finish wiring up the game to use workspaces when signed in to v2 accounts (one of the major new features of v2 accounts) and then I need to update a few technical google-y things and then I'll start rolling it out. I'm hoping next week at some point, though I can't promise for sure.

**Comment by RyanAuxil24** [comment]
Ok Eric.. Hope it comes successfully..
Waiting!🤟🤟

**Comment by vishal332008** [comment]
>To get started, comment below with an introduction of yourself

 ### 👋 Hello everyone!

My name is Vishal and I am 14 years old. I am from India. I have been doing programming from the age of 12. I know python , java , javascript , html , css , react native and basics of c++. 

I have been playing bombsquad for more than a year. I like to mod bombsquad and to find bugs and eliminate them.

Thank You.

**Comment by RyanAuxil24** [comment]
**Hello There!**
(keep in mind that this info is edited, last edited on June 15 2026)

My name is Ryan. I am 18 years old and I am from India.

I'm a programmer, knows Python, HTML, C++, and a video editor. I even make music when bored. I came here for the latest updates, information about BombSquad.
I started playing BombSquad when I was 11 years old, on my PC and an Android TV by connecting the BS remote app. I played with my friends sometimes (even me and my friends loved the game very much).

During the COVID-19 Pandemic, I felt so bored and was reminded about this game again. every day after having any meal, I would go, open my PC/mobile and play this game. Still right now, I still play this and have so much fun! My friends and family really loved playing this...
 
Well, I am the main contributor of the Tamil language in BombSquad. I've contributed this language when I was 13, as this language is my mother language... The Tamil language was added on November 2021 in the version 1.6.6 which was not released on Play Store/App Store. It was released globally with the version 1.7.0 in 2022.

I want to thank Eric for making the game cool, awesome and adding this language to the game throughout the world!

Thank You!

**Comment by xujianing12** [comment]
HELLO EVERY ONE!!!
My English name is Eric, same as efoemling ; ) ,I am 12 years old and I am from china.
I have been doing programming from the age of 9. I know python, javascript, and c++.
I have been playing bombsquad since 2021. I like to mod bombsquad too.
Thank You!

**Comment by MW2917** [comment]
hi.
so I'm also a teenager (16 now) and is kinda new to programming (too new to be commenting here too)
Am studying computer sci+physics+IT+Maths, knows something about python, and can somethings make some pc mod.
I actually played bombsquad like 5 or 6 years ago, but stopped playing for quite a while. It's only half a year since I replayed BS, and a bit more than a month to bs modding.

Not much of a super genius (Unlike comments above), and hopefully playing this game could somewhat help with my com subjects. I think it should, as long as I don't only play bombsquad all the time (don't get me wrong this game is cool af)

**Comment by MatiasMGz** [comment]
### Hi, everyone!
I am Matthew but you can call me Matt. I am a teenager who loves computing and i had played this game since i was a kid.
I am the principal translator for the Spanish language in BombSquad. 
That's all for now!
Bye!
